### PR TITLE
Adding in a canvas drawing checker

### PR DESCRIPTION
**Pull request by jonathanKingston**
image

@Charlie-belmer FYI also: https://github.com/duckduckgo/privacy-test-pages/pull/23

### CODE REVIEW

**Comment by jonathanKingston** [comment]
The code picks a random colour to ensure that we check different channels etc.

File: features/canvas-draw.html [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                        this.context.beginPath();
+                        this.context.arc(x, y, radius, start, end);
+                        this.context.fill();
+                        this.context.beginPath();
+                        this.context.moveTo(x, y);
+                   	}
+                  break;
+                case 'mouseup':
+      	            this.dragging = false;
+      	            this.context.beginPath();
```
Comment (kdzwinel) [inline features/canvas-draw.html:103]: too many spaces!

File: features/canvas-draw.html [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                    this.replicate()
```
Comment (kdzwinel) [inline features/canvas-draw.html:104]: no semicolon! maybe we should merge that inline-eslint first :)

**Review by kdzwinel — COMMENTED** [review]
So cool I have bunch of feature requests :)

1. can we put both DataUrl's one after another and add some `&nbsp;`s to align them so that those hashes are easy to compare (same for ImageData)
1. can we add an indicator when hashes match/don't match (can be just text saying that)
1. can we add a button that always draws the same thing so that we can test if hashes change between domains / reloads ?
1. for integration tests - can we output results to some global variable (e.g. `results`) and have a way of triggering draw (previous point) automatically (e.g. `?run` in url).

**Comment by jonathanKingston** [comment]
>  can we put both DataUrl's one after another and add some &nbsp;s to align them so that those hashes are easy to compare (same for ImageData)

Done, used elements.

> can we add an indicator when hashes match/don't match (can be just text saying that)

Done

> for integration tests - can we output results to some global variable (e.g. results) and have a way of triggering draw (previous point) automatically (e.g. ?run in url).

I added a text box for now, unfortunately the image data is huge and I'm not sure if anything else is meaningful. I guess I could use toDataURL and inject it into an image perhaps.

> can we add a button that always draws the same thing so that we can test if hashes change between domains / reloads ?

Will add tomorrow :)

**Comment by jonathanKingston** [comment]
@kdzwinel ready for your review again.... the run argument will provide default happiness to your day.

image

**Comment by kdzwinel** [comment]
thumbs-up

Perfection ✨

**Pull request merged**